File: src/main/java/frc/robot/util/Rect2d.java
package frc.robot.util;

import edu.wpi.first.math.geometry.Pose2d;

public class Rect2d implements Cloneable {
    private final Pose2d bottomLeftCorner;
    private final Pose2d topRightCorner;

    public Pose2d getBottomLeftCorner() {
        return bottomLeftCorner;
    }
    public Pose2d getTopRightCorner() {
        return topRightCorner;
    }

    public Rect2d(Pose2d bottomLeftCorner, Pose2d topRightCorner) {
        this.bottomLeftCorner = bottomLeftCorner;
        this.topRightCorner = topRightCorner;
    }

    public boolean contains(Pose2d p) {
        return (
                p.getX() > bottomLeftCorner.getX() &&
                        p.getY() > bottomLeftCorner.getY() &&
                        p.getX() < topRightCorner.getX() &&
                        p.getY() < topRightCorner.getY()
        );
    }

    @Override
    public Rect2d clone() {
        return new Rect2d(bottomLeftCorner, topRightCorner);
    }
}

File: src/main/java/frc/robot/util/Rect2d.kt
package frc.robot.util

import edu.wpi.first.math.geometry.Pose2d

class Rect2d(var bottomLeftCorner: Pose2d, var topRightCorner: Pose2d) {
    fun contains(p: Pose2d): Boolean {
        return (
                p.x > bottomLeftCorner.x
                        && p.y > bottomLeftCorner.y
                        && p.x < topRightCorner.x
                        && p.y < topRightCorner.y
                )
    }
}
